### PR TITLE
fix: Backport fixes to v1.8

### LINKED ISSUE #14078: Get the CI on v1.8 green
At the moment, the CI is red on v1.8. See https://github.com/magma/magma/pull/13997#issuecomment-1259054348 for known issues but there might be more problems.

Backport the necessary commits (or create a new PR if it's not easy to backport).

Optional: Enable weekly CI run if it is easy. Discuss if we want to have it.

- [ ] Run the CI on v1.8. Make sure the CI is actually running and not skipped because of path filters.
- [ ] Get all the jobs running on a pull request on v1.8 clean (excluding flakiness and non-required stuff like coverage that is also broken on master).
- [ ] Discuss how to handle this in the future. Create a follow-up ticket, have a discussion in devops meeting...
- [ ] Prepare internal review meeting

### PR DESCRIPTION
**Pull request by MoritzThomasHuebner**
Closes #14078 

## Summary

- Backport https://github.com/magma/magma/pull/13946 and https://github.com/magma/magma/pull/13960.
- Resolve a version issue involving libsystemd-dev that makes the build_all workflow by manually removing a libsystemd0 file from the artifactory cc @nstng 

The issue appears to be a version conflict between `libsystemd0` and `libsystemd-dev`. During the docker build the following error appears:

```
#10 3.365 Reading state information...
#10 3.419 openssl is already the newest version (1.1.1f-1ubuntu2.16).
#10 3.419 openssl set to manually installed.
#10 3.419 python3.8 is already the newest version (3.8.10-0ubuntu1~20.04.5).
#10 3.419 python3.8 set to manually installed.
#10 3.419 Some packages could not be installed. This may mean that you have
#10 3.419 requested an impossible situation or if you are using the unstable
#10 3.419 distribution that some required packages have not yet been created
#10 3.419 or been moved out of Incoming.
#10 3.419 The following information may help to resolve the situation:
#10 3.419 
#10 3.419 The following packages have unmet dependencies:
#10 3.469  libsystemd-dev : Depends: libsystemd0 (= 245.4-4ubuntu3.17) but 245.4-4ubuntu3.18 is to be installed
#10 3.476 E: Unable to correct problems, you have held broken packages.
------
executor failed running [/bin/sh -c apt-get -y update && apt-get -y install     curl     libc-ares2     libev4     libffi-dev     libjansson4     libjemalloc2     libssl-dev     libsystemd-dev     magma-nghttpx=1.31.1-1     net-tools     openssl     pkg-config     python-cffi     python3-pip     python3.8     python3.8-dev     redis-server     iptables     git     automake     gcc     libtool     libcap-ng-dev     linux-headers-generic     netcat     iputils-ping     bcc-tools]: exit code: 100

ERROR: for pipelined  (<Service: pipelined>, 'Build failed')
ERROR: Service 'pipelined' failed to build : Build failed
```

In the 1.8 release, libsystemd-dev is installed from the artifactory and libsystemd0 is installed from the ubuntu archive. On master, both packages are installed from the ubuntu archive. libsystemd0 3.17 has been removed from the ubuntu archive.

There were multiple options to resolve the libsystemd-dev version issue
1. Also upload libsystemd0 3.17 to v1.8 artifactory -> Version might be hard to come by
2. Pin libsystemd-dev to >3.17. We tried this initially, but this does not work since `apt` does not prioritize `>` so much that it looks into a lower priority repository.
3. Remove the magma-preferences file from /etc/apt/preferences.d -> Might have implications that are hard to predict.
4. Remove libsystemd-dev 3.17 from v1.8 artifactory -> This is what we went for

------

I also cherry-picked @nstng 's PR #14176. From his summary:

This change is only relevant for 1.8. - the bazel tests are failing on the 1.8 branch. This is because we are using the latest bazel base image and remote caches that were updated after the 1.8 branch was created. In general it makes sense for a release branch to

    pin the bazel base image to the sha that was latest at branch creation time
    use a dedicated cache key in order to avoid clashes with cache entries that have changes that are not encoded in the cache hash (e.g., links to system libraries are not encoded)

## Test Plan

- [x] Make build_all workflow green.
- [x] Check if CWF integ tests are green. 
- [x] Check if FeG integ tests are green.

https://github.com/MoritzThomasHuebner/magma/actions/runs/3415161087
https://github.com/MoritzThomasHuebner/magma/actions/runs/3415163072

## Additional Information

~~Follow up in https://github.com/magma/magma/issues/14122 for the failing Bazel Build and Test workflow.~~ This is now part of this PR.

### CODE REVIEW

**Review by wolfseb — APPROVED** [review]
lgtm

**Review by tmdzk — APPROVED** [review]

File: lte/gateway/release/build-magma.sh [hunk 1]
```diff
@@ -123,7 +123,7 @@ MAGMA_DEPS=(
     "python3-pip"
     "python3-apt" # The version in pypi is abandoned and broken on stretch
     "python3-aioeventlet" # The version in pypi got deleted
-    "libsystemd-dev"
+    "libsystemd-dev >= 245.4-4ubuntu3.18"
```
Comment (voisey) [inline lte/gateway/release/build-magma.sh]: Out of interest, why do you fix the version everywhere else but have `>=` here?

File: .devcontainer/bazel-base/Dockerfile [hunk 2]
```diff
@@ -48,7 +48,7 @@ RUN echo "Install general purpose packages" && \
         libsctp-dev \
         libssl-dev \
         # dependency of pip systemd
-        libsystemd-dev \
+        libsystemd-dev=245.4-4ubuntu3.18 \
```
Comment (jheidbrink) [inline .devcontainer/bazel-base/Dockerfile]: The PR title says "Backport fixes to v1.8", but this doesn't seem to be on master.

Comment (crasu) [reply .devcontainer/bazel-base/Dockerfile]: Why do we need exactly need this version?

File: lte/gateway/release/build-magma.sh (hunk 1 above)
Comment (MoritzThomasHuebner) [reply lte/gateway/release/build-magma.sh]: I wasn't entirely sure about the reasoning here either, but every other dependency that has a restricted version uses `>=`. I am not even sure if `fpm` wants a `==` or a single `=` for a fixed version.

File: .devcontainer/bazel-base/Dockerfile (hunk 2 above)
Comment (MoritzThomasHuebner) [reply .devcontainer/bazel-base/Dockerfile]: This is due to an issue with the package present on the artifactory.  `libsystemd-dev` currently uses the `245.4-4ubuntu3.17` from the artifactory. It needs the matching version for `libsystemd0`, which is not on the artifactory and has been removed from the ubuntu archive. 


https://launchpad.net/ubuntu/+source/systemd
https://artifactory.magmacore.org/ui/packages/deb:%2F%2Flibsystemd-dev?name=libsystemd&type=packages

File: lte/gateway/release/build-magma.sh (hunk 1 above)
Comment (MoritzThomasHuebner) [reply lte/gateway/release/build-magma.sh:126]: Also not, all the other version fixes are in `apt-get install` commands, which do not accept a range.

File: .devcontainer/bazel-base/Dockerfile (hunk 2 above)
Comment (nstng) [inline .devcontainer/bazel-base/Dockerfile]: bazel-base should not be adapted - a pre-build image is pulled in CI.

File: .github/workflows/agw-workflow.yml [hunk 3]
```diff
@@ -85,7 +85,7 @@ jobs:
           mkdir -p /var/tmp/test_results
           mkdir -p /var/tmp/codecovs
           sudo -E apt-get update -y
-          sudo -E apt-get install -y libsystemd-dev pkg-config curl zip unzip net-tools
+          sudo -E apt-get install -y libsystemd-dev=245.4-4ubuntu3.18 pkg-config curl zip unzip net-tools
```
Comment (nstng) [inline .github/workflows/agw-workflow.yml]: Should be pulled from ubuntu artifactory, i.e., does not need to be pinned. (This is PR workflow, i.e. should be testable on this PR)

File: cwf/gateway/docker/go/Dockerfile [hunk 4]
```diff
@@ -22,7 +22,7 @@ RUN apt-get update && apt-get install -y \
     libjansson-dev \
     libjemalloc-dev \
     libssl-dev \
-    libsystemd-dev \
+    libsystemd-dev=245.4-4ubuntu3.18 \
```
Comment (nstng) [inline cwf/gateway/docker/go/Dockerfile]: magma-preferences is not used in this Dockerfile, i.e., should not have a need to be pinned

File: feg/gateway/docker/go/Dockerfile [hunk 5]
```diff
@@ -25,7 +25,7 @@ RUN apt-get update && apt-get install -y \
     libjansson-dev \
     libjemalloc-dev \
     libssl-dev \
-    libsystemd-dev \
+    libsystemd-dev=245.4-4ubuntu3.18 \
```
Comment (nstng) [inline feg/gateway/docker/go/Dockerfile]: magma-preferences is not used in this Dockerfile, i.e., should not have a need to be pinned

File: feg/gateway/docker/python/Dockerfile [hunk 6]
```diff
@@ -82,7 +82,7 @@ RUN apt-get -y update && apt-get -y install \
   libjansson4 \
   libjemalloc2 \
   libssl-dev \
-  libsystemd-dev \
+  libsystemd-dev=245.4-4ubuntu3.18 \
```
Comment (nstng) [inline feg/gateway/docker/python/Dockerfile]: magma-preferences is not used in this Dockerfile, i.e., should not have a need to be pinned

File: lte/gateway/docker/services/python/Dockerfile [hunk 7]
```diff
@@ -54,7 +54,7 @@ RUN apt-get update && apt-get install -y \
   openjdk-8-jre-headless \
   openjdk-8-jdk \
   pkg-config \
-  libsystemd-dev \
+  libsystemd-dev=245.4-4ubuntu3.18 \
```
Comment (nstng) [inline lte/gateway/docker/services/python/Dockerfile]: magma-preferences is not used in this Dockerfile, i.e., should not have a need to be pinned

File: lte/gateway/release/build-magma.sh (hunk 1 above)
Comment (nstng) [reply lte/gateway/release/build-magma.sh]: Generally magma-preferences is not setup on systems where magma.deb is installed. Imho we do not need this here. Consequence is that magma.deb is not installable on magma-dev - as this needs some fidling anyway, I think this is ok for v1.8 -> should not be pinned

**Comment by crasu** [comment]
@MoritzThomasHuebner  Have you tried if the cwag, feg, agw and orc8r at starts after changing this many dependencies.

**Review by crasu — CHANGES_REQUESTED** [review]
Code itself looks fine but we should test if all these version upgrades do not break anything

File: cwf/gateway/docker/python/Dockerfile [hunk 8]
```diff
@@ -60,7 +60,7 @@ RUN apt-get -y update && apt-get -y install \
     libjansson4 \
     libjemalloc2 \
     libssl-dev \
-    libsystemd-dev \
+    libsystemd-dev>245.4-4ubuntu3.17 \
```
Comment (jheidbrink) [inline cwf/gateway/docker/python/Dockerfile]: I think a small explanation why this is necessary would be nice.

Comment (jheidbrink) [reply cwf/gateway/docker/python/Dockerfile]: (as a comment in the Dockerfile)

**Review by jheidbrink — APPROVED** [review]

**Comment by voisey** [comment]
@MoritzThomasHuebner Is this ready for review? I'm a bit confused by the status of the test plan. If it's ready, can you please request re-reviews from the people who have previously taken a look? Thanks!

**Review by sebathomas — APPROVED** [review]

**Review by crasu — COMMENTED** [review]
As far as I see this is still missing green integ tests

**Review by crasu — APPROVED** [review]

**Comment by crasu** [comment]
@MoritzThomasHuebner Did you double check that cwag and feg start after these new changes now?

**Review by nstng — APPROVED** [review]

**Comment by MoritzThomasHuebner** [comment]
https://github.com/MoritzThomasHuebner/magma/actions/runs/3415161087
https://github.com/MoritzThomasHuebner/magma/actions/runs/3415163072

**Pull request merged**